src/formState.ts

```diff
@@ -236,18 +238,18 @@ export type ObjectFieldConfig<U> = {
 export function createObjectState<T>(
   config: ObjectConfig<T>,
   instance: T,
-  opts: { onBlur?: () => void } = {},
+  opts: { maybeAutoSave?: () => void } = {},
```

Comment:
I'm wondering if this is the right prop name. From the implementers perspective would this be better named `onAutoSave`? Is it still a "maybe" if they're providing it? Internally in formState we'd still have `field.maybeAutoSave()`, but its only a maybe to form-state because it could be a `noop`.

Reply:
I guess when `createObjectState` is being called from `useFormState`, then this naming makes sense.. because we don't know then... But if this function is called directly, as is in a couple places in Blueprint, then it doesn't feel like the right name.

Reply:
Ah yeah...I think the "maybe"-ness of it is that, currently, the caller should still check "is the form changed & dirty & valid" before actually invoking a save...

Right now that `if changed & valid` logic is inside of `useFormState`.

I did consider logic out of `useFormState` into like a `maybeCallOnSave` helper method, which would then only call the caller's callback if those were always true...

But seemed like a bigger/future refactoring...

Reply:
oh yea, true. I didn't think of the instance where a user just blurs out of a field without changing it, so the implementer needs to do their own dirty checks in that case. Would be nice to consolidate that logic in one helper method as you said, but probably fine for now.

Reply:
Agreed we should do that... I'll merge this for now, but if I get time today, will poke at a follow up...